src/main/java/tdl/s3/RemoteSync.java

```diff
@@ -15,49 +15,54 @@
 import tdl.s3.sync.FolderScannerImpl;
 import tdl.s3.sync.FolderSynchronizer;
 import tdl.s3.sync.Source;
+import tdl.s3.sync.SyncProgressListener;
 import tdl.s3.upload.FileUploadingService;
 
 public class RemoteSync {
-    
-    //TODO: Refactor this
-    private static final List<String> FILTERED_EXTENSIONS = Collections.singletonList(".lock");
 
     private final Source source;
-    
+
     private final Destination destination;
-    
+
     private FileUploadingService fileUploadingService;
-    
+
     private FolderSynchronizer folderSynchronizer;
 
+    private SyncProgressListener listener;
+
     public RemoteSync(Source source, Destination destination) {
         this.source = source;
         this.destination = destination;
     }
-    
+
+    public void setListener(SyncProgressListener listener) {
+        this.listener = listener;
+    }
+
     public void run() {
         buildUploadingService();
-        if (source.isUpload()) {
+        if (source.isUpload()) { //TODO: UNUSED
             File file = source.getPath().toFile();
             fileUploadingService.upload(file);
         } else if (source.isSync()) { // just in case sync is not just checking if it's directory
```

Comment:
so if the source is file, should we just throw exception? that would require us to modify the tests, right?